devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1612,6 +2865,64 @@ def clearStateForSlot(codeID) {
 	state["setname$codeID"] = null
 }
 
+/**
+ * Clears the daily repeating start time, end time and weekday bit mask from the state basis the code slot number
+ *
+ * @param codeID: The code slot number
+ */
+def clearDailyRepeatingScheduleDataFromState(codeID) {
+	state["setDailyRepeatingStartTime$codeID"] = null
```

Comment:
A couple of things about this style of state storage:

Why not just store `state["code$codeID"]` as a dict?

I think that would make adding/removing it a lot simpler.

Also I'd check to see that both of these steps are necessary. It would be my instinct that just calling `state.remove` should get rid of the data stored in the field without you having to specifically null it out.